### PR TITLE
[fix]: [CI-5673]: add cache intel limit to hosted, default disabled

### PR DESCRIPTION
**Pull request by jamie-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40678)

### CODE REVIEW

**Comment by jamie-harness** [comment]
trigger messagecheck

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/CIStepGroupUtils.java [hunk 1]
```diff
@@ -121,19 +121,21 @@ public List<ExecutionWrapperConfig> createExecutionWrapperWithInitializeStep(Int
     List<ExecutionWrapperConfig> initializeExecutionSections = new ArrayList<>();
     boolean gitClone = RunTimeInputHandler.resolveGitClone(integrationStageConfig.getCloneCodebase());
     Caching caching = integrationStageConfig.getCaching();
-    boolean saveCache = caching != null && RunTimeInputHandler.resolveBooleanParameter(caching.getEnabled(), true);
+    boolean saveCache = caching != null && RunTimeInputHandler.resolveBooleanParameter(caching.getEnabled(), false);
     CICacheIntelligenceConfig cacheIntelligenceConfig = ciExecutionServiceConfig.getCacheIntelligenceConfig();
     boolean featureCacheEnabled = featureFlagService.isEnabled(CI_CACHE_INTELLIGENCE, accountId);
+    boolean isHosted = infrastructure.getType().equals(Infrastructure.Type.HOSTED_VM);
```
Comment (hemanthmantri) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/CIStepGroupUtils.java:127]: Also for `KUBERNETES_HOSTED` right?

Comment (jamie-harness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/CIStepGroupUtils.java:127]: Just VM now. Having some secret issue with k8s even in hosted

**Review by hemanthmantri — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40678#-:-NHnBoh2A-OvgCdPHdNI:bnfp4nl)* status: 1 file(s) to review. 1. unresolved discussions.

**Review by hemanthmantri — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40678)* status: 1. unresolved discussions.

**Review by hemanthmantri — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40678)* status: :shipit: complete! ✓

**Pull request merged**